### DISCUSSION TITLE
Supabase Auth not working on Mobile Devices

### DISCUSSION THREAD

**Discussion by juanantoniosantanamedina** [supabase/supabase#1894] [Questions]
Good morning.

I'm moving an ionic application from firebase to supabase. It is all working fine while testing in the browser, but when I test it on a mobile device emulator (iOS or Android) the auth API does not work. 
A piece of code as simple as this:

```javascript
async loginOnClick() {
    console.log('Calling auth.signIn');
    //const { user, session, data, error } = await this.supabaseClient.auth.signIn({ email: 'myemail@domain.co.uk', password: 'password' });
    //console.log('User', user);
    //console.log('Data', data);
    //console.log('Session', session);
    //console.log('Error', error);
    this.supabaseClient.auth.signIn({ email: 'myemail@domain.co.uk', password: 'password' })
      .then(({ user, session, data, error }) => {

        console.log('User', user);
        console.log('Data', data);
        console.log('Session', session);
        console.log('Error', error);
      });
}
```

As I said testing in the browser using either the await or the Promise construct works great and it logs the received data, but on the mobile the lines below the call to signIn are not executed. The application does not hang but does nothing, it is like if the function ends there.

Any clue?

Thanks in advance
Juan Antonio Santana

**Comment by egor-romanov** [comment]
My guess would be because you are not awaiting `this.supabaseClient.auth.signIn` and therefore everything in `then`. Could you please check your awaits once again?
